src/libarena/lib.rs

```diff
@@ -188,12 +195,14 @@ impl<T> TypedArena<T> {
         unsafe {
             // Clear the last chunk, which is partially filled.
             let mut chunks_borrow = self.chunks.borrow_mut();
-            let last_idx = chunks_borrow.len() - 1;
-            self.clear_last_chunk(&mut chunks_borrow[last_idx]);
-            // If `T` is ZST, code below has no effect.
-            for mut chunk in chunks_borrow.drain(..last_idx) {
-                let cap = chunk.storage.cap();
-                chunk.destroy(cap);
+            if chunks_borrow.len() > 0 {
+                let last_idx = chunks_borrow.len() - 1;
+                self.clear_last_chunk(&mut chunks_borrow[last_idx]);
```

Comment:
Why not `chunks_borrow.last_mut()`? It might conflict with the drain below, in which case you can use `split_at_mut` I think.